pkg/scheduler/framework/interface.go

```diff
@@ -608,6 +613,36 @@ type Handle interface {
 	Parallelizer() parallelize.Parallelizer
 }
 
+// PreFilterResult wraps needed info for scheduler framework to act upon PreFilter phase.
+type PreFilterResult struct {
+	// The set of nodes that should be considered downstream; if nil then
+	// all nodes are eligible.
+	NodeNames sets.String
+}
+
+func (p *PreFilterResult) Merge(in *PreFilterResult) *PreFilterResult {
+	if p == nil && in == nil {
+		return nil
+	}
+
+	r := PreFilterResult{}
```

Comment:
Why we need to construct a new `PreFilterResult` other than update p directly?

Reply:
this is a common pattern, see the sets package.

Reply:
Got it. Thanks.